src/components/KCodeBlock/KCodeBlock.vue

```diff
@@ -5,207 +5,150 @@
-          color="currentColor"
-          data-testid="k-code-block-is-processing-icon"
-          icon="spinner"
-        />
-
-        <button
-          v-if="query !== ''"
-          appearance="outline"
-          class="k-clear-query-button"
-          data-testid="k-code-block-clear-query-button"
-          title="Clear query"
-          type="button"
-          @click="clearQuery"
+      <Transition name="kongponents-fade-transition">
+        <div
+          v-if="isProcessing || searchQuery"
+          class="code-block-search-results-container"
         >
-          <span class="visually-hidden">Clear query</span>
-
-          <KIcon
-            class="k-clear-icon"
-            :class="[`theme-${theme}`]"
-            color="currentColor"
-            data-testid="k-code-block-clear-icon"
-            icon="clear"
-            :size="KUI_ICON_SIZE_40"
-          />
-        </button>
-      </div>
+          <Transition name="kongponents-fade-transition">
+            <ProgressIcon
+              v-if="isProcessing"
+              class="code-block-processing-icon"
+              :color="getIconColor"
+              data-testid="code-block-processing-icon"
+              :size="KUI_ICON_SIZE_30"
+            />
+          </Transition>
 
-      <div class="k-search-actions">
-        <!-- temporary workaround to get state to "stick" -->
-        <KButton
-          :appearance="isRegExpMode ? 'secondary' : 'tertiary'"
-          :aria-pressed="isRegExpMode"
-          class="k-regexp-mode-button"
-          data-testid="k-code-block-regexp-mode-button"
-          size="small"
+          <p
+            v-if="searchQuery"
+            class="code-block-search-results"
+          >
+            <template v-if="matchingLineNumbers.length === 0">
+              No results
+            </template>
+
+            <template v-else-if="typeof currentLineIndex === 'number' && !isShowingFilteredCode">
+              {{ currentLineIndex + 1 }} of {{ matchingLineNumbers.length }}
+            </template>
+
+            <template v-else>
+              {{ matchingLineNumbers.length }} {{ matchingLineNumbers.length === 1 ? 'result' : 'results' }}
+            </template>
+          </p>
+        </div>
+      </Transition>
+
+      <div class="code-block-search-actions">
+        <CodeBlockIconButton
+          :active="isRegExpMode"
+          :aria-label="`Use regular expression (${ALT_SHORTCUT_LABEL}+R)`"
+          class="regexp-mode-button"
+          data-testid="regexp-mode-button"
+          :theme="theme"
           :title="`Use regular expression (${ALT_SHORTCUT_LABEL}+R)`"
-          type="button"
           @click="toggleRegExpMode"
         >
-          <span class="visually-hidden">RegExp mode enabled</span>
-
-          .*
-        </KButton>
-
-        <!-- temporary workaround to get state to "stick" -->
-        <KButton
-          :appearance="isFilterMode ? 'secondary' : 'tertiary'"
-          :aria-pressed="isFilterMode"
```

Comment:
**Issue**: This being removed is a regression. The buttons that currently have the `aria-pressed` state are toggle buttons and _need_ this to properly expose their pressed state to the accessibility tree.